test/unit-tests/function/probability/lgamma.test.js

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable no-loss-of-precision */
+
+// computation ref: https://www.wolframalpha.com/input?i=LogGamma%5Bx%5D
+
+import assert from 'assert'
+import approx from '../../../../tools/approx.js'
+import math from '../../../../src/defaultInstance.js'
+const lgamma = math.lgamma
+
+const EPSILON = 1e-10
+// lgamma for complex numbers is not quite as precies as lgamma for reals,
+// because `complex.js sin` doesn't have extremely high precision
+const CEPSILON = 5e-8
+
+describe('lgamma', () => {
+  it('should calculate the lgamma of 0 and negative numbers', () => {
+    assert.strictEqual(lgamma(0), Infinity)
+
+    assert.ok(isNaN(lgamma(-0.0005)))
+    assert.ok(isNaN(lgamma(-0.5)))
+    assert.ok(isNaN(lgamma(-1)))
+    assert.ok(isNaN(lgamma(-1.5)))
+    assert.ok(isNaN(lgamma(-2)))
+    assert.ok(isNaN(lgamma(-2.5)))
+    assert.ok(isNaN(lgamma(-100000)))
+    assert.ok(isNaN(lgamma(-123456.123456)))
+  })
+
+  it('should calculate the lgamma of a positive numbers', () => {
+    approx.equal(lgamma(/**/ 0.000000001), /*  */ 20.7232658363691954921, EPSILON)
```

Comment:
Please add here (and below with the complex examples, if it's different) a comment with a reference for the source of the "correct" target values in these tests, thanks.